src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Product/CatalogPriceRule/FormTheme/catalog_price_rules.html.twig

```diff
@@ -0,0 +1,37 @@
+{# **
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * #}
+
+{% extends '@PrestaShop/Admin/TwigTemplateForm/prestashop_ui_kit_base.html.twig' %}
+
+{% block catalog_price_rules_widget %}
+  <div id="catalog-price-rules-container">
+    {{ block('catalog_price_rules_list') }}
+    {{ form_widget(form) }}
+  </div>
+{% endblock %}
+
+{% block catalog_price_rules_list %}
+  {% include '@PrestaShop/Admin/Sell/Catalog/Product/CatalogPriceRule/Blocks/list.html.twig' %}
+{% endblock %}
```

Comment:
```suggestion
{% block catalog_price_rules_widget %}
  <div {{ block('widget_container_attributes') }}>
    {% include '@PrestaShop/Admin/Sell/Catalog/Product/CatalogPriceRule/Blocks/list.html.twig' %}
  </div>
{% endblock %}
```

Reply:
Little explanation of the suggestions:
- `block('widget_container_attributes')` is used to automatically add the widget attributes (ID, attributes from the `attr`, ...) of a widget, it's slightly different from `block('widget_attributes')` because it's mainly for div containers not final input fields (so it doesn't include the `disabled` `required` or attributes that have no impact on a div) It's a very convenient block to know and use when creating a custom widget, because this way the form type can still be heavily configured via its options
- `{{ form_widget(form) }}` is not needed here since the form has no children
- the `block catalog_price_rules_list` was overkill since it's never reused, so you can add the `include` directly in the widget layout

Reply:
Nice, didn't know about widget_container_attributes!